From: Cixi Geng <cixi.geng1@unisoc.com>
Subject: checkpatch: add judgment condition for Kconfig help test
Date: Thu, 16 Nov 2023 23:39:04 +0800

has_help only counts the situation in which the patch file adds a help
line.  If the config was renamed and the description was modified,
has_help is zero because a "help" line was not added.

here is one case:
    the modify file: drivers/iio/adc/Kconfig  line(1047)

    -config SC27XX_ADC
    +config SPRD_ADC
     	tristate "Spreadtrum SC27xx series PMICs ADC"
     	depends on MFD_SC27XX_PMIC || COMPILE_TEST
    +	depends on ARCH_SPRD
     	help
    -	  Say yes here to build support for the integrated ADC inside the
    -	  Spreadtrum SC27xx series PMICs.
    +	  Say yes here to build support for the integrated ADC inside of the
    +	  Spreadtrum SC27xx and UMPxx series PMICs.

     	  This driver can also be built as a module. If so, the module
     	  will be called sc27xx_adc.

the checkpatch result:
    WARNING: please write a help paragraph that fully describes the config symbol
    #23: FILE: drivers/iio/adc/Kconfig:1050:
    +config SPRD_ADC
    +	tristate "Spreadtrum's ADC PMICs driver"
     	depends on MFD_SC27XX_PMIC || COMPILE_TEST
    +	depends on ARCH_SPRD
     	help
    +	  Say yes here to build support for the integrated ADC inside of the
    +	  Say yes here to build support for the integrated ADC inside of the
    +	  Say yes here to build support for the integrated ADC inside of the
    +	  Spreadtrum SC27xx and UMPxx series PMICs.

     	  This driver can also be built as a module. If so, the module
     	  will be called sc27xx_adc.

    total: 0 errors, 1 warnings, 17 lines checked

Link: https://lkml.kernel.org/r/20231116153904.15589-1-cixi.geng@linux.dev
Fixes: b8709bce9089 ("checkpatch: improve Kconfig help test")
Signed-off-by: Cixi Geng <cixi.geng1@unisoc.com>
Cc: Dwaipayan Ray <dwaipayanray1@gmail.com> (reviewer:CHECKPATCH)
Cc: Joe Perches <joe@perches.com>
Cc: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Cc: Randy Dunlap <rdunlap@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 scripts/checkpatch.pl |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/scripts/checkpatch.pl~checkpatch-add-judgment-condition-for-kconfig-help-test
+++ a/scripts/checkpatch.pl
@@ -3616,7 +3616,7 @@ sub process {
 					$needs_help = 1;
 					next;
 				}
-				if ($f =~ /^\+\s*help\s*$/) {
+				if ($f =~ /^\+\s*help\s*$/ || $f =~ /^\s*help\s*$/) {
 					$has_help = 1;
 					next;
 				}
_
